File: src/types/recommendation.ts
export type RecommendationLevel = 'recommended' | 'tryable' | 'unsupported' | 'unknown';

export interface Recommendation {
  id: string;
  title: string;
  level: RecommendationLevel;
  backend_id?: string | null;
  estimated_latency_ms?: number | null;
  required_actions: string[];
  launch_profile_id?: string | null;
}

export type LaunchConfig = Record<string, string | number | boolean>;

export interface LaunchResult {
  ok: boolean;
  message: string;
}

File: src-tauri/src/models/recommendation.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Recommendation {
    pub id: String,
    pub title: String,
    pub level: String,
    pub backend_id: Option<String>,
    pub estimated_latency_ms: Option<u128>,
    pub required_actions: Vec<String>,
    pub launch_profile_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LaunchResult {
    pub ok: bool,
    pub message: String,
}
